File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/tag/domain/Tag.java
package io.twogether.nbe_5_7_2_02team.tag.domain;

import io.twogether.nbe_5_7_2_02team.post.domain.PostTag;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;

import java.util.ArrayList;
import java.util.List;

@Entity
public class Tag {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @OneToMany(mappedBy = "tag", fetch = FetchType.LAZY)
    private List<PostTag> postTags = new ArrayList<>();

    public Tag() {}

    public Tag(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public List<PostTag> getPostTags() {
        return postTags;
    }
}

File: NBE_5_7_2_02TEAM/src/main/java/io/twogether/nbe_5_7_2_02team/tag/domain/Tag.kt
package io.twogether.nbe_5_7_2_02team.tag.domain

import io.twogether.nbe_5_7_2_02team.post.domain.PostTag
import jakarta.persistence.*

@Entity
class Tag (
    @field:Column(nullable = false)
    var name: String,
) {
    @field:Id
    @field:GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
}
